File: admin/src/main/java/com/project/admin/controller/dto/response/PageResponse.java
package com.project.admin.controller.dto.response;

import org.springframework.data.domain.Page;
import java.util.List;

public record PageResponse<T>(
        List<T> content,
        Pagination  pagination

) {
    public  record Pagination(
            long totalCount,
            int pageSize,
            int currentPage
    ){}

    public static <T> PageResponse<T> of(Page<T> page, int currentPage) {

        return new PageResponse<>(
                page.getContent(),
                new Pagination(
                        page.getTotalPages(),
                        page.getSize(),
                        currentPage
                )
        );
    }

}

File: api/src/main/kotlin/com/project/api/controller/dto/PageResponse.kt
package com.project.api.controller.dto

import com.project.api.service.dto.PageResult

data class PageResponse(
    val totalCount: Long,
    val pageSize: Int,
    val currentPage: Int,
) {
    companion object {
        fun from(pageResult: PageResult) = PageResponse(
            totalCount = pageResult.totalCount,
            pageSize = pageResult.pageSize,
            currentPage = pageResult.currentPage,
        )
    }
}